WIP: Add frontmatter for SEO and integrate with SEO component

Shelob9 opened the pull request:
This PR addresses #7 

@gvan12 I added a description to the frontmatter for patterns, and updated the query in patterns template, and made changes to SEO and TemplateWrapper component. Can you please let me know if this approach makes sense?  I wanted to make sure I was on the right path before going to far.

Shelob9 committed:
Pass post frontmatter to TemplateWrapper and SEO component #7

Shelob9 commented on src/components/SEO.js:
```diff
@@ -5,233 +5,239 @@ import Helmet from "react-helmet";
+	const URL = `${homeURL}${postPath || ""}`;
+	const image = `${homeURL}${config.siteBanner}`;
+
+	if (article) {
```
@gvan12 What other meta fields should be supported?

gvan12 replied:
We could support additional images for pages by adding that to the frontmatter and using a fallback of the siteBanner. This will be more useful once we actually start using blog posts (currently named news I think). Blogs are _sort of_ supported in that way but beefing up the frontmatter/CMS integration is necessary beforehand.

Shelob9 replied:
> We could support additional images for pages by adding that to the frontmatter and using a fallback of the siteBanner. 
Makes sense

What would the frontmatter look like:
```md
image2: /assets/whatever.jpg
image3: /assets/iamg3.jpg
```
?

gvan12 replied:
> > We could support additional images for pages by adding that to the frontmatter and using a fallback of the siteBanner.
> > Makes sense
> 
> What would the frontmatter look like:
> 
> ```lisp
> image2: /assets/whatever.jpg
> image3: /assets/iamg3.jpg
> ```
> 
> ?

I think we'd just have 1 image per page - so it'd pick that one first, then if none existed fall back to the `siteBanner`

Shelob9 replied:
I added a featuredimage field as that's the name used in "News"

Shelob9 commented on src/components/SEO.js:
```diff
@@ -5,233 +5,239 @@ import Helmet from "react-helmet";
+					url: `${homeURL}${config.siteLogo}`
+				}
+			},
+			//datePublished: postNode.parent.birthtime,
```
@gvan12 I commented these out to prevent errors. I can't figure out how to adjust the query so this is included.

gvan12 reviewed (commented):
I think this is a great direction. Make sure you add the corresponding fields in the CMS files to allow editors to change them. Also - I added prettier to a husky pre-commit hook so these styling changes shouldn't show up in future commits/prs.

Shelob9 commented:
@gvan12 Thanks for the feedback. I will make sure to get the CMS fields.

I can block off a lot of tomorrow to getting this done for all of the pages.

Shelob9 committed:
#7 merge master to branch

Shelob9 committed:
#7 fix formatting in changed files

Shelob9 committed:
Add SEO fields for patterns #7

Shelob9 committed:
#7 impliment SEO for patterns

Shelob9 committed:
Add SEO related fields for all pages to cms config.yml #7

Shelob9 committed:
update queries to include SEO fields #7

Shelob9 committed:
pass post to layout in templates #7

Shelob9 commented:
@gvan12 I have now added:

- support in SEO component for description, keywords, and featuredimage fields.
- Declared description, keywords, and featuredimage fields for all pages in netlify cms yml
- Updated propTypes for SEO and Layout component to specify above fields as optional props.

I did not:

- Change the preview components - **These fields are not visible. I think I need to add a widget for this?** @gvan12 This is the one place I'm stuck. If you can give me some direction on what's required here -- this is my first experience with netlify CMS -- I can finish this off.
- Figure out how to login to the CMS from localhost.

gvan12 commented:
Thanks for your work on this @Shelob9 ! I'll nudge it over the finish-line.
